framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -119,344 +97,390 @@ private static Pattern anchored(String regex) {
      * @param localScope path to local scope to use
      * @param useLocalScope whether {@code localScope} should be used to resolve identifiers
      */
-    public static FlowExpressions.Receiver parse(
+    public static Receiver parse(
             String expression,
             FlowExpressionContext context,
             TreePath localScope,
             boolean useLocalScope)
             throws FlowExpressionParseException {
         context = context.copyAndSetUseLocalScope(useLocalScope);
-        FlowExpressions.Receiver result = parseHelper(expression, context, localScope);
+        Expression expr;
+        try {
+            expr = StaticJavaParser.parseExpression(replaceParameterSyntax(expression));
+        } catch (ParseProblemException e) {
+            throw constructParserException(expression, "is an invalid expression");
+        }
+        Receiver result = getReceiverFromExpression(expr, context, localScope);
         if (result instanceof ClassName && !expression.endsWith("class")) {
             throw constructParserException(
                     expression, "a class name cannot terminate a flow expression string");
         }
         return result;
     }
 
-    private static FlowExpressions.Receiver parseHelper(
-            String expression, FlowExpressionContext context, TreePath path)
+    /**
+     * Converts the passed Javaparser Expression to a {@link Receiver}.
+     *
+     * @return the Receiver of the passed expression
+     */
+    private static Receiver getReceiverFromExpression(
```

Comment:
This method should just be removed.  The calls to it in the visitor can be replaced with `expr.accept(this, context)`.